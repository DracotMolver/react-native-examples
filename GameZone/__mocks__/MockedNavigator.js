import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';

const Stack = createStackNavigator();

const MockComponent = () => {
  return <div>test</div>;
};

const MockedNavigator = ({ routeName, component, headerTitleComponent }) => (
  <NavigationContainer>
    <Stack.Navigator initialRouteName={routeName || 'title'}>
      <Stack.Screen
        name={routeName || 'title'}
        component={component || MockComponent}
        options={{
          headerTitle: () => headerTitleComponent
        }}
      />
    </Stack.Navigator>
  </NavigationContainer>
);

export default MockedNavigator;
