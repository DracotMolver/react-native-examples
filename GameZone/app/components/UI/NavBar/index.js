import React from 'react';
import {
  useWindowDimensions,
  ImageBackground,
  StyleSheet,
  View,
  Text
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const styles = StyleSheet.create({
  header: {
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    height: '100%',
    flex: 1
  },
  headerText: {
    letterSpacing: 1,
    fontWeight: 'bold',
    fontSize: 20,
    color: '#333'
  },
  icon: {
    position: 'absolute',
    left: 8
  }
});

const NavBar = ({ title }) => {
  const { width } = useWindowDimensions();
  const navigation = useNavigation();

  function onPressOpenDrawer() {
    navigation.openDrawer();
  }

  return (
    <ImageBackground
      testID="navbar-imgback"
      source={require('../../../assets/images/game_bg.png')}
      style={{ ...styles.header, width }}>
      <Icon
        testID="navbar-icon"
        onPress={onPressOpenDrawer}
        color="black"
        style={styles.icon}
        size={48}
        name="menu"
      />
      <View testID="navbar-header-container" style={styles.headerText}>
        <Text>{title}</Text>
      </View>
    </ImageBackground>
  );
};

export default NavBar;
