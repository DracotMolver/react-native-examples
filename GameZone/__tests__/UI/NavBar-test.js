import React from 'react';
import { render, cleanup } from 'react-native-testing-library';
// mock
import MockedNavigator from '../../__mocks__/MockedNavigator';
// component
import NavBar from '../../app/components/UI/NavBar';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

afterEach(cleanup);

describe('Testing UI - NavBar component', () => {
  it('Should render right', () => {
    const rendered = render(
      <MockedNavigator headerTitleComponent={<NavBar title="test" />} />
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('Should has two children', () => {
    const { getByTestId } = render(
      <MockedNavigator headerTitleComponent={<NavBar title="test" />} />
    );

    const imgBack = getByTestId('navbar-imgback');

    expect(imgBack.props.children.length).toEqual(2);
  });
});
