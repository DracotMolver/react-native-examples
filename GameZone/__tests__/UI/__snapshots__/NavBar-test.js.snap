// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UI - NavBar component Should render right 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 64,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <div>
                      test
                    </div>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(224, 224, 224)",
                "flex": 1,
                "shadowColor": "rgb(224, 224, 224)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": 750,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/game_bg.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": 750,
                      },
                      undefined,
                    ]
                  }
                  testID="navbar-imgback"
                />
                <Text
                  allowFontScaling={false}
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 48,
                      },
                      Object {
                        "left": 8,
                        "position": "absolute",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="navbar-icon"
                >
                  î—’
                </Text>
                <View
                  style={
                    Object {
                      "color": "#333",
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "letterSpacing": 1,
                    }
                  }
                  testID="navbar-header-container"
                >
                  <Text>
                    test
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
